sdlib/d/gc/tcache.d

```diff
@@ -78,23 +160,37 @@ public:
 				copySize = getSizeFromClass(oldSizeClass);
 			}
 		} else {
-			auto esize = pd.extent.size;
-			if (alignUp(size, PageSize) == esize) {
+			oldFill = pd.extent.allocSize;
+
+			// Prohibit resize below appendable fill:
+			if (size < oldFill)
+				return ptr;
+
+			auto oldSize = oldFill ? oldFill : pd.extent.size;
+
+			if (alignUp(size, PageSize) == alignUp(oldSize, PageSize)) {
```

Comment:
You are short circuiting the whole thing when the size you'd get is the same. Yet, later on, you recompute the size.